tests/contrib/operators/test_gcs_to_s3_operator.py

```diff
@@ -72,3 +72,32 @@ def test_execute(self, mock_hook, mock_hook2):
                          sorted(uploaded_files))
         self.assertEqual(sorted(MOCK_FILES),
                          sorted(hook.list_keys('bucket', delimiter='/')))
+
```

Comment:
I think we need at least one more test - where the list of files differ and we send a small subset of files.

Reply:
That test is already there from line 48 to 73, the test upload the first file into S3 and then sync all gcs existing files, and the expected behaviour is to, after running the operator, all the files exists in the s3 bucket. The unit test that is missing is the one that set replace as True and all the files are replaced. I can add it, just confirm that the existing tests, cover what are you talking about @potiuk please

Reply:
Yeah. I see the test but it's not exactly what I talked about. I think we have two cases that are "extremes". The first test uses "replace=True" - so it will work no matter if the original file was there or not and always copies all files.  The second test tests that nothing has been copied because the files are already there. What I talked about is something in the middle - I think it might be pretty useful to test the "incremental" behaviour in case of replace = False where one file is there, but some other file is not - and we only copy the missing file.

Reply:
That's exactly the test that is already present:

lines 62:66
```Python
       # create dest bucket
        hook = S3Hook(aws_conn_id=None)
        b = hook.get_bucket('bucket')
        b.create()
        b.put_object(Key=MOCK_FILES[0], Body=b'testing') # here is the incremental happening
```
Crate the destination bucket and create just the first element of the MOCK_FILES  list in the S3 bucket (**note the [0] in line 66**), so now the destination bucket have
`TEST1.csv`
and the origin bucket have all the files in the MOCK_FILES
`TEST1.csv, TEST2.csv, TEST3.csv`

Line 71:73
```Python
        uploaded_files = operator.execute(None)
        self.assertEqual(sorted(MOCK_FILES[1:]),
                         sorted(uploaded_files))
```
the **uploaded_files** variable have the files that were uploaded from gcs to s3 due the execution of the operator, in the line 72 we expect that the files in MOCK_FILES, except the first one, were transfer (**note the [1:] in line 72**), because it was already there

and then in the line 74, expect that all the files that are in the origin, exists in the destination.

Reply:
Aaaaah. I see now. The first test originally did not have replace value set and the second had replace = False so somehow I assumed that replace was TRUE by default (which it was not). So I thought the "incremental" behaviour was not tested :). What was not tested was replace=True in fact. Which I see you just added ! Cool. Also the docstring now helps with explaining that it is False. Maybe to avoid confusion you could add replace in all cases or remove it whenever it is set to False (default). It might be misleading otherwise.

Great tests! It looks really good now.

Reply:
Changes made, now all the test have the replace param.